janis_bioinformatics/tools/io_lib/scramble/base.py

```diff
@@ -0,0 +1,228 @@
+from abc import ABC
+from typing import Any, Dict
+
+from janis_bioinformatics.data_types import Bam, Cram, FastaFai, FastqGzPair, File
+from janis_bioinformatics.tools.bioinformaticstoolbase import BioinformaticsTool
+from janis_core import (
+    Boolean,
+    CaptureType,
+    CpuSelector,
+    Filename,
+    Float,
+    InputSelector,
+    Int,
+    Stdout,
+    String,
+    ToolArgument,
+    ToolInput,
+    ToolMetadata,
+    ToolOutput,
+    get_value_for_hints_and_ordered_resource_tuple,
+)
+
+SCRAMBLE_MEM_TUPLE = [
+    (
+        CaptureType.key(),
+        {
+            CaptureType.TARGETED: 8,
+            CaptureType.EXOME: 12,
+            CaptureType.CHROMOSOME: 12,
+            CaptureType.THIRTYX: 16,
+            CaptureType.NINETYX: 20,
+            CaptureType.THREEHUNDREDX: 24,
+        },
+    )
+]
+
+SCRAMBLE_CORES_TUPLE = [
+    (
+        CaptureType.key(),
+        {
+            CaptureType.TARGETED: 16,
+            CaptureType.EXOME: 20,
+            CaptureType.CHROMOSOME: 24,
+            CaptureType.THIRTYX: 30,
+            CaptureType.NINETYX: 32,
+            CaptureType.THREEHUNDREDX: 32,
+        },
+    )
+]
+
+
+class ScrambleBase(BioinformaticsTool, ABC):
+    def tool(self):
+        return "scramble"
+
+    def friendly_name(self):
+        return "scramble"
+
+    def tool_provider(self):
+        return "io_lib"
+
+    def base_command(self):
+        return ["scramble"]
+
+    def inputs(self):
+        return [
+            ToolInput("inputFilename", Bam(), position=200),
+            ToolInput(
+                "reference", FastaFai(), prefix="-r", doc="Reference sequence file.",
+            ),
+            ToolInput("outputFilename", Filename(extension=".bam")),
+            *ScrambleBase.additional_inputs,
+        ]
+
+    def arguments(self):
+        return [
+            ToolArgument("bam", prefix="-I", doc="input data format",),
+            ToolArgument("cram", prefix="-O", doc="output data format",),
+            ToolArgument(
+                "-9",
+                Boolean(),
```

Comment:
Hey @matthdsm, a `ToolArgument` doesn't have a type. your line can just be:

```python
ToolArgument(
    "-9",
    doc="compression settings for output cram file (-1=fast,-9=best)",
)
```

Otherwise, I'm happy with this PR after this little fix if you want me to merge?

Reply:
Fixed!